eng/packages/http-client-csharp-mgmt/emitter/src/resolve-arm-resources-converter.ts

```diff
@@ -313,46 +264,21 @@ function getModelFromSdkContext(
 }
 
 /**
- * Helper to get method ID from an operation using the operation maps.
- * Tries identity-based lookup first, then path-based, then name-based lookup.
+ * Get method ID (cross-language definition ID) from an Operation.
+ * Uses TCGC's getCrossLanguageDefinitionId for reliable lookup.
  */
 function getMethodIdFromOperation(
   operation: Operation,
-  operationPath: string,
-  operationMaps: {
-    byIdentity: Map<Operation, SdkMethod<SdkHttpOperation>>;
-    byName: Map<string, SdkMethod<SdkHttpOperation>>;
-    byPath: Map<string, SdkMethod<SdkHttpOperation>[]>;
-  }
+  sdkContext: CSharpEmitterContext
 ): string | undefined {
```

Comment:
please update the signatures in this file to ensure if they have sdkContext parameter, it should always be the first one .